Allows filling fire extinguishers with beakers/glass containers.

emmanuelbassil opened the pull request:
🆑 emmanuelbassil
tweak: Can now fill fire extinguishers using beakers. Can effectively fill them with any liquid.
tweak: Fire extinguishers have a chance to break when firing any liquid that isn't water. They cannot be repaired.
tweak: Borgs can repair broken fire extinguishers at the recharging station.
/🆑 

Putting in any acid will melt the extinguisher and have it spray you.
Also changed the prob (50) for using an acid dispenser into a guaranteed self-spray. The idea that an exploding fire extinguisher has a 50% chance of gently dropping some liquid on you is weird. At least get a full spray to the mouth.

emmanuelbassil commented:
As per discord feedback, extinguisher is already better than chemsprayer with the limited reagents it accepts.
Added 20% chance to break per spray if the extinguisher contains anything that isn't water, compensating for the ability to now put anything into an extinguisher. Also makes chem sprayers much more valuable now.
A 'broken' extinguisher has a gunked up spraying mechanism and doesn't fire anymore, cannot be repaired.

Question is, should I allow acid in small amounts to be placed in extinguishers now or do I keep the total ban on acid?

SierraKomodo commented on code/game/objects/items/weapons/extinguisher.dm:
```diff
@@ -79,41 +81,68 @@
+		qdel(src)
+		return TRUE
+
+	to_chat(user, SPAN_NOTICE("You fill \the [src] with [amount] units from \the [target]."))
```
```suggestion
	to_chat(user, SPAN_NOTICE("You fill \the [src] with [amount] unit\s from \the [target]."))
```

SierraKomodo commented on code/game/objects/items/weapons/extinguisher.dm:
```diff
@@ -79,41 +81,68 @@
-		return
+			return TRUE
+
+		to_chat(user, SPAN_NOTICE("You fill \the [src] with [amount] units from \the [container]."))
```
```suggestion
		to_chat(user, SPAN_NOTICE("You fill \the [src] with [amount] unit\s from \the [container]."))
```

SierraKomodo commented on code/modules/mob/living/silicon/robot/modules/_module.dm:
```diff
@@ -205,6 +205,12 @@
 				P.loaded += new P.ammo_type(src)
 
+	var/obj/item/extinguisher/extinguisher = locate() in equipment
+	if (extinguisher && extinguisher?.broken)
```
```suggestion
	if (extinguisher?.broken)
```

SierraKomodo commented on code/modules/mob/living/silicon/robot/modules/_module.dm:
```diff
@@ -205,6 +205,12 @@
 			if (length(P.loaded) <= P.max_shells)
 				P.loaded += new P.ammo_type(src)
 
+	var/obj/item/extinguisher/extinguisher = locate() in equipment
+	if (extinguisher && extinguisher?.broken)
+		var/remaining_volume = extinguisher.reagents.total_volume
+		extinguisher.broken = FALSE
+		extinguisher.reagents.remove_any(remaining_volume)
+
```
Shouldn't this also re-add the water the extinguisher normally spawns with?

SierraKomodo replied:
Disregard, per discord convo.

SierraKomodo commented on code/modules/reagents/Chemistry-Holder.dm:
```diff
@@ -246,6 +246,15 @@ GLOBAL_DATUM_INIT(temp_reagents_holder, /obj, new)
 
+/datum/reagents/proc/has_other_reagent(list/check_reagents)
+	for (var/datum/reagent/current in reagent_list)
+		if (current.type in check_reagents)
+			continue
+		else
+			return TRUE
+
```
```suggestion
        if (!(current.type in check_reagents))
            return TRUE

```

emmanuelbassil commented:
Concerns handled

SierraKomodo commented on code/game/objects/items/weapons/extinguisher.dm:
```diff
@@ -21,6 +21,8 @@
 	var/safety = 1
 	var/sprite_name = "fire_extinguisher"
+	var/list/preferred_reagent = list(/datum/reagent/water)
+	var/broken
```
```suggestion
	var/broken = FALSE
```

emmanuelbassil committed:
Fill extinguishers with glass containers

Fixes

Auto stash before rebase of "refs/heads/FireBeaker"

Auto stash before rebase of "refs/heads/FireBeaker"

Fixes

Auto stash before rebase of "refs/heads/FireBeaker"

Auto stash before rebase of "refs/heads/MoreAttackby"

Auto stash before rebase of "refs/heads/FireBeaker"

Auto stash before rebase of "refs/heads/FireBeaker"

Fixes

emmanuelbassil commented:
Fixed!

Spookerton commented on code/game/objects/items/weapons/extinguisher.dm:
```diff
@@ -79,41 +81,68 @@
+			return TRUE
+
+		to_chat(user, SPAN_NOTICE("You fill \the [src] with [amount] unit\s from \the [container]."))
+		playsound(src, 'sound/effects/pour.ogg', 25, 1)
```
avoid 1/0 where TRUE/FALSE are appropriate
